### PR TITLE
feat(maintner)!: gRPC API is only supported API

### LINKED ISSUE #4: maintner: move stack to gRPC
The REST api for maintner has worked well, but for internal consistency we should move to gRPC. Note this would entail removing the `rtr` package.

### PR DESCRIPTION
**Pull request by orthros**
Removes the "apiroutes" directory in maintnerd as the REST API is no longer
supported.

Adds gRPC health checking to maintner and maintner-rtr

The cloud build functionality now uses the API descriptor set for the
service definition as opposed to a swagger spec.

BREAKING_CHANGE: The "v0" API for maintner is no longer supported. The
"v1" API is still supported via HTTP => gRPC transcoding done by the
Extensible Service Proxy (ESP).

Fixes #4

### CODE REVIEW

**Comment by jmdobry** [comment]
Let me know when ready for review

**Comment by orthros** [comment]
@jmdobry Ready to review

**Comment by jmdobry** [comment]
Does anybody use the v0 API?

**Comment by orthros** [comment]
I'm fairly confident they do not; I just checked access logs for the past 7 days and there are no requests to the v0 API.

**Comment by orthros** [comment]
And fwiw sloth uses v1 https://github.com/googleapis/sloth/blob/e13c62189639dbe6d513358a16119126eaea02f2/src/issue.ts#L54

File: drghs/endpoints/maintnerd.yaml [hunk 1]
```diff
@@ -13,203 +13,15 @@
...
-consumes:
-- "application/json"
-produces:
-- "application/json"
-schemes:
-# Uncomment the next line if you configure SSL for this API.
-#- "https"
-- "http"
-paths:
-  "/update":
```
Comment (jmdobry) [inline drghs/endpoints/maintnerd.yaml]: With this path gone, DevRel Services is unable to update the repo list, that call is currently failing on DEV

Comment (orthros) [reply drghs/endpoints/maintnerd.yaml]: Yep; this is known

**Comment by orthros** [comment]
Thanks for the tests!

This is tested on dev, correct?

The ones that are only returning closed issues I think have to do with the changes to filtering I have on my machine that are not published to this branch yet. I'll review.

As for the `grpc: received message larger than max (5109322 vs. 4194304))` errors. Those are present in prod as well (without these changes) as that repository has way too many Issues/PRs. There is a separate PR that adds pagination which will fix those errors

**Comment by jmdobry** [comment]
Yes I was testing that on DEV.

Let me know if you deploy a fix for the `closed` filtering.

Ack on the pagination, yes that will fix it.

Can you add integration tests in this PR that make and verify all those same requests to our test repo (https://github.com/GoogleCloudPlatform/repo-automation-playground)? I can go to that repo and add test Issue/PR data to cover the use cases.

**Comment by orthros** [comment]
I can add those integration tests, but it'll be a bit cumbersome depending on what you're looking for. 

Would you prefer these tests spinning up a fresh instance of maintner, using a user-provided api key, building the log from scratch and then makes api calls against itself and verifies the results? Or would you prefer an integration test that has a "blessed" mutation log that is uploaded to a user-specified gcs bucket for maintner to read from and then the test harness to make api calls against?

The former will require developers to have their own GCP project with a bucket, and github api key, the latter will require "only" the GCP project/bucket, but will require some code changes to maintnerd to ensure it does not begin a syncloop.

Personally,  I prefer the second as it will be faster and not eat up rate limit.

**Comment by orthros** [comment]
A follow up to my earlier point: having the files be checked into the repository would help keep the tests stable

**Comment by orthros** [comment]
FWIW I've updated the unit tests on the filters for the API

**Comment by jmdobry** [comment]
I like the second idea, but that seems too much for this PR.

However it's done, I think there needs to be automated test that in some shape or form performs the same tests that I did manually in that table above.

**Comment by jmdobry** [comment]
`closed=false` is still returning closed issues on DEV

**Comment by orthros** [comment]
Thanks for letting me know; I hadn't deployed the latest changes.

**Comment by orthros** [comment]
Added the Update RPC

File: drghs/endpoints/maintnerd.yaml (hunk 1 above)
Comment (orthros) [reply drghs/endpoints/maintnerd.yaml]: This should be resolved with the new gRPC method

**Comment by jmdobry** [comment]
The tests I did above all seem to be passing now (except for the include comments one, that one will be fixed by switching to pagination)

**Comment by jmdobry** [comment]
Any new changes that need to be pushed to this PR?

**Pull request merged**